### PR TITLE
Only examine non-unknown server descriptions to determine incompatibi…

### PR DESCRIPTION
**Pull request by p-mongo**
…lity

Without this, a replica set topology where the client received a response from primary and is waiting for a response from secondary is considered incompatible, since the secondary's wire version is taken to be 0.

https://jira.mongodb.org/browse/RUBY-1583

### CODE REVIEW

**Comment by p-mongo** [comment]
I am wondering if only data bearing servers should be examined, analogously with logical session timeout. The current implementation in Ruby driver considers all non-unknown servers as specified in this PR.

**Comment by ajdavis** [comment]
I think that Unknown servers should *not* be considered but arbiters *should* be considered when the driver is determining whether it is compatible with the topology. This change LGTM. Is there a prose change we should make in the SDAM spec?

**Comment by ajdavis** [comment]
RSOther and RSGhost members might be real members of the replica set, or they might be on their way to become members. So let's consider their wire versions when determining driver compatibility. Considering all server types but Unknown is simple and consistent with some of the other spec rules.

So, the existing test and prose change LGTM. You told me in a meeting that there are associated Ruby new spec tests that are incoming, let's add them to this PR>

**Comment by p-mongo** [comment]
Added tests for arbiters, ghosts and others.

**Comment by p-mongo** [comment]
Spec ticket: https://jira.mongodb.org/browse/SPEC-1172

File: source/server-discovery-and-monitoring/tests/rs/incompatible_ghost.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+description: "Compatible topology with incompatible ghost"
```
Comment (ajdavis) [inline source/server-discovery-and-monitoring/tests/rs/incompatible_ghost.yml]: Remember to update: "Incompatible"

File: source/server-discovery-and-monitoring/tests/rs/incompatible_other.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{
+  "description": "Compatible topology with incompatible other",
```
Comment (ajdavis) [inline source/server-discovery-and-monitoring/tests/rs/incompatible_other.json]: Incompatible

**Pull request merged**